posthog/src/main/java/com/posthog/java/PostHog.java

```diff
@@ -194,4 +199,43 @@ private JSONObject getEventJson(String event, String distinctId, Map<String, Obj
         }
         return eventJson;
     }
+
+    /**
+     * 
+     * @param featureFlag which uniquely identifies your feature flag
+     *
+     * @param distinctId which uniquely identifies your user in your database. Must
+     *                   not be null or empty.
+     */
+    public boolean getFeatureFlag(String featureFlag, String distinctId) {
+        return getFeatureFlags(distinctId).get(featureFlag) != null;
+    }
+
+    /**
+     * 
+     * @param featureFlag which uniquely identifies your feature flag
+     *
+     * @param distinctId which uniquely identifies your user in your database. Must
+     *                   not be null or empty.
+     */
+    public String getFeatureFlagPayload(String featureFlag, String distinctId) {
+        return getFeatureFlags(distinctId).get(featureFlag);
+    }
+
+    private HashMap<String, String> getFeatureFlags(String distinctId) {
+        JSONObject response = sender.post("/decide/?v=3", distinctId);
+
+        HashMap<String, String> featureFlags = new HashMap<>();
```

Comment:
Sorry, maybe I'm not being 100% clear here.

`/decide` returns a json of the form:

```
{
featureFlagPayloads: { flag-one: 'payload-one', flag-two: 'payload-two'},
featureFlags: {flag-one: true, flag-two: false, flag-three: 'variant-value'}
}
```

Now, from the java SDK, we expect:

`getFeatureFlag('flag-one', 'distinctID')` to return `true`
`isFeatureEnabled('flag-one', 'distinctID')` to return `true`
`getFeatureFlagPayload('flag-one', 'distinctID')` to return `payload-one`, also might be a good idea to parse this payload since it's sent as a JSON string.


`getFeatureFlag('flag-three', 'distinctID')` to return `variant-value`
`isFeatureEnabled('flag-three', 'distinctID')` to return `true`
`getFeatureFlagPayload('flag-three', 'distinctID')` to return null.


So, effectively, getFeautreFlag returns whatever decide sends us in the `featureFlags` dict.

getFeatureFlagPayload returns whatever is in the featureFlagPayloads dict for that flag.

and isFeatureEnabled returns a boolean value of `getFeatureFlag`

Reply:
Does this make sense?

Reply:
other than that, just missing tests for these, and we should be good to go :)

Reply:
Make total sense, I'll fix that and send to you

Reply:
@neilkakkar done. Check if I understand your appointments?